client-js/app/elements/labrad-registry.ts

```diff
@@ -42,15 +43,76 @@ export class LabradRegistry extends polymer.Base {
                                               this.$.editValueInput);
   }
 
+
+  getDefaultSelectedItem(): number {
+    return this.path.length > 0 ? 1 : 0;
+  }
+
+
+  cursorMove(e) {
+    const offset = this.path.length > 0 ? 1 : 0;
+    const length = this.keys.length + this.dirs.length + offset;
+
+    switch (e.detail.combo) {
+      case 'up':
+        if (this.selectedIdx === null || this.selectedIdx === 0) {
+          this.set('selectedIdx', null);
+        } else {
+          this.set('selectedIdx', this.selectedIdx - 1);
+        }
+        break;
+
+      case 'down':
+        if (this.selectedIdx === null) {
+          this.set('selectedIdx', 0);
+        } else if (this.selectedIdx < length - 1) {
+          this.set('selectedIdx', this.selectedIdx + 1);
+        }
+        break;
+
+      default:
+        // Nothing to do.
+        break;
+    }
+  }
+
+
+  cursorTraverse(e) {
+    if (this.selectedIdx === null) {
+      return;
+    }
+
+    const item = this.$.combinedList.selectedItem;
+    const link = item.querySelector('a');
+
+    // If we have a link, we want to traverse down.
+    if (link) {
+      this.fire('app-link-click', {path: link.path});
+    }
+  }
+
+
+  cursorBack(e) {
+    if (this.path.length === 0) {
+      return;
+    }
+
+    const parentPath = this.path.slice(0, -1);
+    const parentUrl = this.places.registryUrl(parentPath);
+    this.fire('app-link-click', {path: parentUrl});
+  }
+
+
   /**
-   * on a path change, we deselect everything, empty filterText
+   * On path change, select the default element and clear the filter.
    */
   @observe('path')
   pathChanged(newPath: string[], oldPath: string[]) {
-    this.selectedIdx = null;
-    this.filterText = '';
+    this.set('selectedIdx', this.getDefaultSelectedItem());
+    this.set('filterText', '');
```

Comment:
It's too bad we have to go through these polymer APIs since we lose typesafety...